site/src/components/address-input/address-input.tsx

```diff
@@ -31,252 +23,97 @@ export class AddressInput {
    @Prop() public lookupDelay: number;
 
-   @State() private addressValue: string;
-   @State() private address2Value: string;
    @State() private zipValue: string;
    @State() private cityValue: string;
    @State() private countyValue: string;
    @State() private stateValue: string;
-   /**
-    * Address options returned from SmartyStreets
-    */
-   @State() private addressOptions: {
-      text: string,
-      suggestion: USAutocomplete.Suggestion,
-   }[];
-
-   private addressSelect?: HTMLSelectElement;
-   private addressInput?: HTMLInputElement;
-   private smartyStreetsLookupTimeout?: number;
+   @State() private cityOptions: Set<string>;
+   @State() private countyOptions: Set<string>;
+   @State() private stateOptions: Map<string, string>;
 
    constructor() {
       this.lookupDelay = 200;
-      this.showStateSelect = false;
-      this.addressValue = "";
-      this.address2Value = "";
       this.zipValue = "";
       this.cityValue = "";
       this.countyValue = "";
       this.stateValue = "";
-      this.addressOptions = [];
+      this.cityOptions = new Set();
+      this.countyOptions = new Set();
+      this.stateOptions = new Map();
    }
 
    public render() {
+      const zipValidationRegex = /^\d{5}$/;
 
-      const states = States;
-
-      /**
-       * When user alters address input, lookup address using SmartyStreets on a small delay to prevent thrashing
-       */
-      const addressOnInput = ( event: Event ) => {
-         const value = ( ( event.target as HTMLInputElement ).value || "" ).trim();
+      const onZipInputChange = (event: Event) => {
+         this.zipValue = ( event.target as HTMLSelectElement ).value;
 
-         clearTimeout( this.smartyStreetsLookupTimeout );
-
-         this.addressValue = value;
-         // don't do lookup if user hasn't entered anything into the address input
-         if( value !== "" ) {
-            this.smartyStreetsLookupTimeout = window.setTimeout( () => {
-               if( this.smartyStreetsApiKey != null ) {
-                  this.getSuggestions( this.addressValue );
-               } else {
-                  console.log( "No SmartyStreets API key provided, cannot lookup addresses" );
+         if(zipValidationRegex.test(this.zipValue)) {
+            ZipGeocode(this.zipValue).
+            then((result) => {
+               if(result.error) {
+                  console.log(result.error);
+                  return;
                }
-            }, this.lookupDelay );
-         } else {
-            this.addressOptions = [];
-         }
-      };
 
-      /**
-       * Focus the address selection list when user presses keydown with address input focused
-       */
-      const addressOnKeyDown = ( event: KeyboardEvent ) => {
-         if( this.addressOptions.length > 0 && event.key === "ArrowDown" ) {
-            event.stopImmediatePropagation();
-            event.preventDefault();
-            this.addressSelect?.focus();
+               // @ts-ignore
```

Comment:
Is there a better way to do this? Do I need to create a type for the object returned by ZipGeocode?

Reply:
Yes. There should already be one in the SmartyStreets types though. If not, yeah just make a new one and put it there.